src/test/scala/nl.knaw.dans.easy.solr4files/UpdateServletSpec.scala

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.apache.http.HttpStatus._
+import org.apache.solr.client.solrj.response.UpdateResponse
+import org.apache.solr.client.solrj.{ SolrClient, SolrRequest, SolrResponse }
+import org.apache.solr.common.SolrInputDocument
+import org.apache.solr.common.util.NamedList
+import org.scalamock.scalatest.MockFactory
+import org.scalatra.test.scalatest.ScalatraSuite
+
+import scala.util.{ Failure, Success }
+import scalaj.http.HttpResponse
+
+class UpdateServletSpec extends TestSupportFixture
+  with ServletFixture
+  with ScalatraSuite
+  with MockFactory {
+
+  private class StubbedWiring extends ApplicationWiring(new Configuration("", new PropertiesConfiguration() {
+    addProperty("solr.url", "http://deasy.dans.knaw.nl:8983/solr/easyfiles")
+    addProperty("vault.url", s"http://deasy.dans.knaw.nl:20110/")
+  })) {
+    override lazy val solrClient: SolrClient = new SolrClient() {
+      // can't use mock because SolrClient has a final method, now we can't count the actual calls
+
+      override def deleteByQuery(q: String): UpdateResponse =
+        throw createHttpException(SC_BAD_REQUEST)
+
+      override def commit(): UpdateResponse =
+        throw new Exception("not expected call")
+
+      override def add(doc: SolrInputDocument): UpdateResponse =
+        throw new Exception("not expected call")
+
+      override def close(): Unit = ()
+
+      override def request(solrRequest: SolrRequest[_ <: SolrResponse], s: String): NamedList[AnyRef] =
+        throw new Exception("not expected call")
+    }
+  }
+
+  private class App extends EasyUpdateSolr4filesIndexApp(null)
+  private val mockedApp = mock[App]
+  private val stubbedDeleteApp = new EasyUpdateSolr4filesIndexApp(new StubbedWiring)
+  addServlet(new EasyUpdateSolr4filesIndexServlet(mockedApp), "/*")
+  //addServlet(new EasyUpdateSolr4filesIndexServlet(stubbedDeleteApp), "/stubbed-delete/*")
```

Comment:
This breaks a lot of tests so I guessed wrong. I guess it would be similar to implementing another servlet as done in the bag-store. So I hope this can wait for the next PR-chunk for the issue.